### PR TITLE
external-plugins/cherrypicker: add support for branch chains

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR adds support for listing multiple branches to cherrypick to in a single line. When multiple branches are detected, a cherrypick is created for the first listed branch and the cherrypick will contain a `/cherrypick` command for the remaining branches. This allows a user to specify a chain of cherrypicks, where the first listed branch must be merged before moving onto the next.

### CODE REVIEW

File: cmd/external-plugins/cherrypicker/server.go [hunk 1]
```diff
@@ -186,10 +186,15 @@ func (s *Server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent
 	})
 
 	cherryPickMatches := cherryPickRe.FindAllStringSubmatch(ic.Comment.Body, -1)
-	if len(cherryPickMatches) == 0 || len(cherryPickMatches[0]) != 2 {
+	if len(cherryPickMatches) == 0 || len(cherryPickMatches[0]) < 2 {
 		return nil
 	}
-	targetBranch := strings.TrimSpace(cherryPickMatches[0][1])
+	branches := strings.Fields(cherryPickMatches[0][1])
+	targetBranch := branches[0]
```
Comment (stevekuznetsov) [inline cmd/external-plugins/cherrypicker/server.go:193]: Does this array access need a `len` check?

Comment (stevekuznetsov) [reply cmd/external-plugins/cherrypicker/server.go:193]: I guess the match can't be an empty string...

File: cmd/external-plugins/cherrypicker/server_test.go [hunk 2]
```diff
@@ -351,6 +351,7 @@ func testCherryPickIC(clients localgit.Clients, t *testing.T) {
 	if err := s.handleIssueComment(logrus.NewEntry(logrus.StandardLogger()), ic); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+	t.Logf("Comments: %+v; PRs: %+v", ghc.comments, ghc.prs)
```
Comment (stevekuznetsov) [inline cmd/external-plugins/cherrypicker/server_test.go:354]: debug info to remove

File: cmd/external-plugins/cherrypicker/server_test.go [hunk 3]
```diff
@@ -481,6 +488,9 @@ func testCherryPickPR(clients localgit.Clients, t *testing.T) {
 	var expectedFn = func(branch string) string {
 		expectedTitle := fmt.Sprintf("[%s] This is a fix for Y", branch)
 		expectedBody := fmt.Sprintf("This is an automated cherry-pick of #%d", prNumber)
+		if branch == "release-1.3" {
```
Comment (stevekuznetsov) [inline cmd/external-plugins/cherrypicker/server_test.go:491]: I think the new functionality in this PR should support long chains transparently, can you add a test for that?

File: cmd/external-plugins/cherrypicker/server.go (hunk 1 above)
Comment (AlexNPavel) [reply cmd/external-plugins/cherrypicker/server.go:193]: Yeah, the regex requires there to be a space for a match

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**